packages/core/src/2d/text/TextRenderer.ts

```diff
@@ -438,30 +402,162 @@ export class TextRenderer extends Renderer {
     }
   }
 
+  private _resetCharFont(): void {
+    const lastCharFont = this._charFont;
+    if (lastCharFont) {
+      lastCharFont._addRefCount(-1);
+      lastCharFont.destroy();
+    }
+    this._charFont = Font.createFromOS(
+      this.engine,
+      TextUtils.getNativeFontHash(this.font.name, this.fontSize, this.fontStyle)
+    );
+    this._charFont._addRefCount(1);
+  }
+
   private _updatePosition(): void {
```

Comment:
`_updatePosition ()`->`_updateWorldPosition()` is better to express function purpose